### PR TITLE
[GPMR] Support of complex numbers

### PR DESCRIPTION
**Pull request by amontoison**
I need #481 to add tests with complex numbers.

### CODE REVIEW

File: src/gpmr.jl [hunk 1]
```diff
@@ -169,8 +169,8 @@ function gpmr!(solver :: GpmrSolver{T,FC,S}, A, B, b :: AbstractVector{FC}, c ::
   γ ≠ 0 || error("c must be nonzero")
   @. U[1] = c₀ / γ
 
-  # Compute ‖r₀‖² = γ² + β²
-  rNorm = sqrt(γ^2 + β^2)
+  # Compute ‖r₀‖² = |γ|² + |β|²
+  rNorm = sqrt(abs(γ)^2 + abs(β)^2)
```
Comment (amontoison) [inline src/gpmr.jl:173]:
```suggestion
  # Compute ‖r₀‖² = γ² + β²
  rNorm = sqrt(γ^2 + β^2)
```

Comment (amontoison) [reply src/gpmr.jl:173]: I forgot that `γ` and `β` are real numbers...

File: src/gpmr.jl [hunk 2]
```diff
@@ -397,8 +397,8 @@ function gpmr!(solver :: GpmrSolver{T,FC,S}, A, B, b :: AbstractVector{FC}, c ::
       pos = pos - j + 1               # position of rᵢ.ⱼ₋₁
     end
     # Rₖ can be singular is the system is inconsistent
```
Comment (dpo) [inline src/gpmr.jl:399]:
```suggestion
    # Rₖ can be singular if the system is inconsistent
```

**Review by dpo — CHANGES_REQUESTED** [review]

File: src/gpmr.jl (hunk 2 above)
Comment (amontoison) [reply src/gpmr.jl:399]: Thanks !
It's already fixed on the `main` branch (https://github.com/JuliaSmoothOptimizers/Krylov.jl/blob/main/src/gpmr.jl#L404).
The mistake will be removed after the rebase.

**Pull request merged**